Migration docs revamp. (#24630)
dotnet/docs: docs/core/porting/breaking-changes.md

---
title: Breaking changes
description: Breaking changes between releases of .NET and how it impacts compatibility.
author: stebon
ms.date: 06/14/2021
---
# Breaking Changes

Visit the following documentation to learn about and find breaking changes when moving from the .NET Framework to .NET and between releases of .NET.

## [What are breaking changes?](../compatibility/index.md)

## [Types of compatibility.](../compatibility/categories.md)

> ## [Find breaking changes.](../compatibility/breaking-changes.md)

Reviewer: This needs more content. We wouldn't want to link to another article inside of a header. Have the header, but there needs to be more information, like a paragraph explaining the section concept, then use something like:

`For more information, see LINK` which is our typical wording for linking people to more info.

Author: Will do.

Author: Done

> ---
> title: Breaking changes
> description: Breaking changes between releases of .NET and how it impacts compatibility.
> author: stebon
> ms.date: 06/14/2021
> ---
> # Breaking Changes

Reviewer: This is probably too generic and conflicts with the existing content. We don't want the search engines or users confused. I'm having a hard time figuring out what to put here that differentiates this article from others related to breaking changes. Please feel free to change this.

- Title changed
- Description needs to be longer
- `ms.author` is your alias
- `author` is your github id

```suggestion
---
title: Breaking changes can affect porting your app
description: Breaking changes can occur when porting your code from .NET framework to .NET. This article describes the categories of breaking changes you may encounter when porting.
author: StephenBonikowsky
ms.author: stebon
ms.date: 06/14/2021
---
# Breaking changes may occur when porting code
```

Changes that affect compatibility, otherwise known as breaking changes, will occur between versions of .NET. This is particularly impactful when porting from .NET Framework to .NET due to certain technologies not being available as well as the changes needed to make .NET a cross-platform framework.

We strive to maintain a high level of compatibility between .NET versions, so while breaking changes will occur, they are carefully considered.

> Before upgrading major versions check the breaking changes documentation for changes that might impact you.

Reviewer: There should be commas after the intro clauses. Look for these in other sentences too.
```suggestion
Before upgrading major versions, check the breaking changes documentation for changes that might impact you.
```

> ## Categories of breaking changes.

Reviewer: Headers shouldn't end in periods.
```suggestion
## Categories of breaking changes
```

There are several types of breaking changes...

- modifications to the public contract
- behavioral changes
- Platform support
- Internal implementation changes
- Code changes

For details about what is allowed or disallowed see [Changes that affect compatibility](../compatibility/index.md)

> ## Types of compatibility.

Reviewer: Headers shouldn't end in periods.
```suggestion
## Types of compatibility
```

> Compatibility refers to the ability to compile or execute code on a version of a .NET implementation other than the one with which the code was originally developed.

Reviewer: Style guide info related to the word `execute`: https://styleguides.azurewebsites.net/Styleguide/Read?id=2700&topicid=33512
```suggestion
Compatibility refers to the ability to compile or run code on a version of a .NET implementation other than the one with which the code was originally developed.
```

> For details on the six different ways a change can affect compatibility see [How code changes can affect compatibility](../compatibility/categories.md)

Reviewer: Maybe it would be good to bullet point these 6?

> ## Find breaking changes.

Reviewer: Headers shouldn't end in periods.

```suggestion
## Find breaking changes
```
